README.md

```diff
@@ -0,0 +1,357 @@
+
+- A user identifier.
+- A resource pattern that identifies to which resources does this rule applies. This pattern is a simple regular expression
+following MQTT rules for pattern matching.
+- A permission type, that can be one of read, write and readwrite.
+
+The user identifier used in each ACL entry corresponds to the AMQP user used to connect to RabbitMQ. For anonymous
+connections, the `'global'` user will be used (take into account that the `'global'` user is defined as an Erlang atom
+instead of an Erlang string, so the `"global"` user could be created and will differ from the former).
+
+For this plugin's shake, resources (represented by routing keys) will be defined as strings, possible separated by the
+`/` character. Only alphanumerical characters are allowed in the routing keys. Each substring between separation characters
+represents a resource level. Resource patterns are resource strings where some resource levels may be replaced by the
+`*` wildcard. A resource pattern string `/level1/level2/.../levelN/#` terminated in the `#` wildcard will match with all
+resources containing the prefix `/level1/level2/.../levelN/`.
+
+The following table shows the meaning of each permission type
+
+| Permissions         | Definition                                     |
+| ------------------- |:---------------------------------------------- |
+| read       	      | User can bind queues to the selected resource. |
+| write       	      | User can publish messages to the selected key. |
+| readwrite       	  | Combination of both read and write permissions.|
+
+##### ACL File format
+
+The following excerpt shows an example of an ACL file:
+
+```
+topic read #
+
+# Topic permissions for Jack Doe user
+user jackdoe
+topic read root/messages
+topic readwrite root/subroot/+
+
+# Topic permissions for Jennifer Doe
+user jenniferdoe
+topic read root/messages
+```
+
+All the lines beginning with the `#` character are ignored as comments.
+
+As it can be seen from the example, the files are divided into different user sections, separated by the `user <username>`
+expression. For each user section, there is a list of `topic <permission> <pattern>` expressions indicating the permissions
+the user have over different resource patterns.
+
+The file starts with a list of topic expressions without a user header. Those expressions define the default permissions
+for every user. Whenever a specific permission cannot be found for a user for a specific routing key, the default permissions
+will be applied. Default permissions are modelled internally as corresponding to the user `'global'`.
+
+If, after applying both the user specific and the global permissions no permissions are found, the access will be rejected.
+
+#####  <a name="aclstore_commands"/> Permission storage
+
+The ACL Store has been developed as a OTP generic server. To ease the use of the server, a set of functions was added to
```

Comment:
Is a standard part of the Erlang VM (something like J2EE is for Java). I'll expand it.